cmake/FindGeant4.cmake

```
@@ -16,7 +16,11 @@ include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Geant4 CONFIG_MODE)
 
 if(Geant4_FOUND AND Geant4_VERSION VERSION_GREATER_EQUAL 11 AND CELERITAS_USE_CUDA)
-  target_compile_features(Geant4::G4global INTERFACE cuda_std_17)
+  if(TARGET Geant4::G4global)
+    target_compile_features(Geant4::G4global INTERFACE cuda_std_17)
+  else()
+    target_compile_features(Geant4::G4global-static INTERFACE cuda_std_17)
+  endif()
```

Since Geant4 can build both static and shared at the same time, maybe we should attach the compile features to both?
```suggestion
  foreach(_tgt Geant4::G4global Geant4::G4global-static)
    if(TARGET ${_tgt})
      target_compile_features(${_tgt} INTERFACE cuda_std_17)
    endif()
  endforeach()
  unset(_tgt)
```

---

Yep, that's definitely a cleaner way to do it - committed!